definitions/npm/react-redux_v5.x.x/flow_v0.62.0-/test_connect.js

```diff
@@ -0,0 +1,280 @@
+// @flow
+import React from "react";
+import { connect } from "react-redux";
+
+function testPassingPropsToConnectedComponent() {
+  type Props = {passtrough: number, passtroughWithDefaultProp: number, fromStateToProps: string};
+  class Com extends React.Component<Props> {
+    static defaultProps = { passtroughWithDefaultProp: 123 };
+    render() {
+      return <div>{this.props.passtrough} {this.props.fromStateToProps}</div>;
+    }
+  }
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passtrough={123} forMapStateToProps={'data'} passtroughWithDefaultProp={123}/>;
+  // OK without passtroughWithDefaultProp
+  <Connected passtrough={123} forMapStateToProps={'data'}/>;
+  //$ExpectError wrong type for passtrough
+  <Connected passtrough={''} forMapStateToProps={'data'} passtroughWithDefaultProp={123}/>;
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passtrough={123} forMapStateToProps={321} passtroughWithDefaultProp={123}/>;
+  //$ExpectError wrong type for  passtroughWithDefaultProp
+  <Connected passtrough={123} forMapStateToProps={'data'} passtroughWithDefaultProp={''}/>;
+  //$ExpectError passtrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passtrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testWithStatelessFunctionalComponent() {
+  type Props = {passtrough: number, fromStateToProps: string};
+  const Com = (props: Props) => <div>{props.passtrough} {props.fromStateToProps}</div>
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
```

Comment:
since you don't have a custom `mergeProps`, `InputProps` (e.g. `forMapStateToProps`) should also be part of component props

Reply:
If I would use exact props then maybe yes, otherwise no. Yes they gets the same props but i'm not interested them in them elsewhere than where i'm using them. That doesn't mean that I _couldn't_ use same props definition in both though. You are free to use `forMapStateToProps` in your component props if you want to, and it will fail if it contradicts with `InputProps`